lib/services/watcher.go

```diff
@@ -159,8 +159,33 @@ func (p *resourceWatcher) Close() {
 	p.cancel()
 }
 
+// IsInitialized is a non-blocking way to check if resource watcher is already
+// initialized.
 func (p *resourceWatcher) IsInitialized() bool {
-	return p.collector.isInitialized()
+	select {
+	case <-p.collector.initializationChan():
+		return true
+	default:
+		return false
+	}
+}
+
+// WaitInitialization blocks until resource watcher is fully initialized with
+// the resources presented in auth server.
+func (p *resourceWatcher) WaitInitialization() error {
+	// wait for resourceWatcher to complete initialization.
+	t := time.NewTicker(500 * time.Millisecond)
+	defer t.Stop()
+	for {
+		select {
+		case <-p.collector.initializationChan():
+			return nil
+		case <-t.C:
+			p.Log.Debugf("ResourceWatcher %s is not yet initialized.", p.collector.resourceKind())
```

Comment:
This will spam debug logs super fast. We should either increase the ticker interval or make sure this is only printed every few seconds.